src/groups/mqb/mqbblp/mqbblp_domain.cpp

```diff
@@ -136,7 +138,40 @@ int validateConfig(bsl::ostream& errorDescription,
         return rc_CHANGED_STORAGE_TYPE;  // RETURN
     }
 
-    return 0;
+    // Validate newConfig.subscriptions()
+
+    bsl::size_t size   = newConfig.subscriptions().size();
+    bool        result = true;
+
+    for (bsl::size_t i = 0; i < size; ++i) {
+        const mqbconfm::Expression& expression =
+            newConfig.subscriptions()[i].expression();
+
+        if (mqbconfm::ExpressionVersion::E_VERSION_1 == expression.version()) {
+            if (expression.text().length()) {
+                bmqeval::CompilationContext context(allocator);
+
+                if (!bmqeval::SimpleEvaluator::validate(expression.text(),
+                                                        context)) {
+                    errorDescription
+                        << "Expression validation failed: [ expression: "
+                        << expression << ", rc: " << context.lastError()
+                        << ", reason: \"" << context.lastErrorMessage()
+                        << "\" ]";
+                    result = false;
+                }
+            }
+        }
+        else {
+            if (expression.text().length()) {
+                errorDescription << "Invalid Expression: [ expression: "
+                                 << expression;
+                result = false;
+            }
+        }
+    }
```

Comment:
This whole block is pretty complex, can we break it down into something more like this?

```c++
for (Subscriptions::const_iterator it = newConfig.subscriptions().cbegin(),
                                                       end = newConfig.subscriptions().cend();
       it != end;
      ++it)
{
    const mqbconfm::Expression& expression = it->expression();
    if (!validateExpression(errorDescription, expression, allocator)) {
        return rc_INVALID_SUBSCRIPTION;
    } 
    return 0;
}

```